src/components/ITSBStore/itsbGraph.js

```diff
@@ -0,0 +1,149 @@
+ * types of nodes:
+ * 
+ * - author (id, name)
+ * - place (GeoJSON/LP feature)
+ * - waypoint (one stop on the itinerary, linked to 
+ *   author, place, citations, and scoped by time
+ * 
+ * Edges:
+ * - waypoints are linked to author
+ * - waypoints are linked to places
+ * - waypoints are (directionally!) linked to waypoints
+ */
+export class ITSBGraph {
+
+  constructor() {
+    this.graph = createGraph();
+  }
+
+  init = (authors, places, itineraries) => {
+    this.graph.clear();
+
+    this.graph.beginUpdate();
+
+    // Authors and places can be treated the same way
+    const authorNodes = authors.map(normalizeNode);
+    const placeNodes = places.map(normalizeNode);
+
+    [...authorNodes, ...placeNodes ].forEach(n => 
+      this.graph.addNode(n.id, n));
+
+    // Itineraries are split to waypoint nodes
+    const waypointNodes = itineraries.reduce((waypoints, it) => {
+      return [...waypoints, ...splitItinerary(it)]
+    }, []).filter(wp => { 
+      const isValid = this.exists(wp.author, wp.place);
+
+      if (!isValid)
+        console.warn('Skipping unconnected node', wp);
+
+      return isValid;
+    });
+
+    waypointNodes.forEach(wp => this.graph.addNode(wp.id, wp));
+
+    // Link waypoints to authors
+    waypointNodes.forEach(wp => {
+      const sourceId = wp.id;
+      const targetId = wp.author;
+      this.graph.addLink(sourceId, targetId, { relation: 'visitedBy' });
+    });
+
+    // Link waypoints to places
+    waypointNodes.forEach(wp => {
+      const sourceId = wp.id;
+      const targetId = wp.place;
+      this.graph.addLink(sourceId, targetId, { relation: 'locatedAt' });
+    });
+
+    // Link waypoints in sequence
+    waypointNodes.reduce((previous, wp) => {
+      if (previous && previous.author === wp.author) {
+        const sourceId = wp.id;
+        const targetId = previous.id;
+        this.graph.addLink(sourceId, targetId, { relation: 'previous'});
+      }
+
+      return wp;
+    }, null);
+
+    this.graph.endUpdate();
+  }
+
+  getNode = id =>
+    this.graph.getNode(id)?.data;
+
+  listNodesWithProperty = (key, value) => {
+    const nodes = [];
+
+    this.graph.forEachNode(n => {
+      if (n.data[key] === value)
+        nodes.push(n.data);
+    });
+
+    return nodes;
+  }
+
+  getLinksOfType = (nodeId, relation) => {
+    const links = [];
+
+    this.graph.forEachLinkedNode(nodeId, (node, link) => {
+      if (link.data.relation === relation)
+        links.push(link);
+    }, false);
+
+    return links;
+  }
+
+  listAuthors = () =>
+    this.listNodesWithProperty('@type', 'Person');
+
+  listPlaces = () =>
+    this.listNodesWithProperty('type', 'Feature');
+
+  listItineraries = (dateRange) => {
+    const allWaypoints = this.listNodesWithProperty('type', 'waypoint');
+
+    const filtered = dateRange ? 
+      filterByTime(allWaypoints, dateRange) : allWaypoints;
+
+    const groupedByAuthor = groupBy(filtered, 'author');
```

Comment:
Another place where maybe things could get out of sequence? Though I would think it should still be in sequence per author, I haven't tested `groupBy` and not sure if it preserves order.

Reply:
I re-read the groupBy function (it was copied from StackOverflow after all ;-) It uses a `reduce` which means it will preserver order, too. (My guess is actually that my time parsing code may be broken somewhere...)